### PR TITLE
fix: add terminationGracePeriodSeconds

### PR DESCRIPTION
**Pull request by olsove**

### CODE REVIEW

**Comment by olsove** [comment]
Do we bump chart version in feature branches manually or is it automated?

File: charts/common/README.md [hunk 1]
```diff
@@ -58,6 +58,7 @@ Highlighted features:
 | service.internalPort | int | 8080 | Set the internal port for your service |
 | shortname | string | `nil` | `id` for GCP 2.0, typically on the form `theapp`. Max 10 characters |
 | team | string | `nil` | Your team name, without a `team-` prefix |
+| terminationGracePeriodSeconds | int | `nil` | Override pod terminationGracePeriodSeconds (default 30s). If set to > 30 it will be used else ignored. |
```
Comment (AlexanderBrevig) [inline charts/common/README.md:61]: Move to `.Values.container.terminationGracePeriodSeconds`

File: charts/common/templates/deployment.yaml [hunk 2]
```diff
@@ -139,6 +139,9 @@ spec:
         {{- toYaml .Values.container.volumes | nindent 8 }}
       {{- end }}
       {{- include "topologyspreadconstraints" . | indent 6 }}
+      {{- if and .Values.terminationGracePeriodSeconds (gt (int .Values.terminationGracePeriodSeconds) 30) }}
```
Comment (AlexanderBrevig) [inline charts/common/templates/deployment.yaml:142]: Maybe allow setting whatever value (disallowing 29 seems arbitrary)

File: charts/common/tests/deployment_test.yaml [hunk 3]
```diff
@@ -261,3 +261,20 @@ tests:
...
           path: spec.template.spec.containers[0].livenessProbe.exec.command
+  - it: must use terminationGracePeriodSeconds if gt than 30
+    set:
+      <<: *values
+      terminationGracePeriodSeconds: 60
+    asserts:
+      - equal:
+          path: spec.template.spec.terminationGracePeriodSeconds
+          value: 60
+  - it: skip use of terminationGracePeriodSeconds if missing or less to 31
```
Comment (AlexanderBrevig) [inline charts/common/tests/deployment_test.yaml:272]: Is there a rationale behind this? Maybe I'm missing something :)

**Review by AlexanderBrevig — CHANGES_REQUESTED** [review]

File: charts/common/tests/deployment_test.yaml (hunk 3 above)
Comment (olsove) [reply charts/common/tests/deployment_test.yaml]: changed to allow any value

**Review by AlexanderBrevig — APPROVED** [review]
LGTM, kanskje endre til fix: så det blir en minor patch?

**Pull request merged**